common/__init__.py

```diff
@@ -7,28 +7,39 @@
 
 
 @functools.cache
-def compute_normalization_factors(impact_defs):
-    """Compute normalization factors for impact definitions.
+def get_normalization_weighting_factors(impact_defs):
+    """Compute normalization and weighting factors for impact definitions.
 
     Args:
         impact_defs: A frozen dictionary of impact definitions. Must be deepfrozen (using frozendict.deepfreeze()) because this function is cached using @functools.cache, which requires immutable arguments.
 
     Returns:
         A frozen dictionary mapping impact keys to their normalization factors.
     """
-    normalization_factors = {
-        "ecs": {
-            k: v["ecoscore"]["weighting"] / v["ecoscore"]["normalization"]
-            for k, v in impact_defs.items()
-            if v["ecoscore"] is not None
-        },
-        "pef": {
-            k: v["pef"]["weighting"] / v["pef"]["normalization"]
-            for k, v in impact_defs.items()
-            if v["pef"] is not None
-        },
-    }
-    return frozendict(normalization_factors)
+    return frozendict(
+        {
+            "ecs_normalizations": {
+                k: v["ecoscore"]["normalization"]
+                for k, v in impact_defs.items()
+                if v["ecoscore"] is not None
+            },
+            "pef_normalizations": {
+                k: v["pef"]["normalization"]
+                for k, v in impact_defs.items()
+                if v["pef"] is not None
+            },
+            "ecs_weightings": {
+                k: v["ecoscore"]["weighting"]
+                for k, v in impact_defs.items()
+                if v["ecoscore"] is not None
+            },
+            "pef_weightings": {
+                k: v["pef"]["weighting"]
+                for k, v in impact_defs.items()
+                if v["pef"] is not None
+            },
+        }
+    )
```

Comment:
It looks like it’s the same code 4 times. What about refactoring it using a lambda or a function?